### DISCUSSION TITLE
[Feature] Sprite Cropping

### DISCUSSION THREAD

**Discussion by Amidweiz** [ikemen-engine/Ikemen-GO#827] [Feature Request] [answered]
This was an old idea that I originally suggested to K4thos over a year ago, not sure if you ever took note of it but might as well tell it again here. Sprite/animation cropping so you're able to selectively crop an opponent or yourself. How I I'd think this implementation of this could go out could either be based off a line you can define the X,Y, and angle values for, or a sprite that acts as a mask if a more precise crop is needed. This could benefit characters that need to remove the opponent only partially (for example basara from samurai shodown's zetsumei ougi where he drags you underground and kills you https://youtu.be/QqQBnYSD60o?t=165 (I think he also has a command grab like this too but I can't remember currently). Current versions in mugen from what I can remember have a large shadow to hide the opponent going underground which the effect goes away when you have a stage that is ground is higher up in the air enough so you can see the opponent underground when you shouldn't.

>[state 123465, Linear Crop]
>type = linearcrop
>trigger1 = whatever
>x = 1 ;where to define the x position of the crop, not sure where it would be defined on the
>opponent at the moment though
>y = 3 ;y position of the crop
>rotation = 30; the angle of the crop measured in radians since mugen does that for angles. I'm
>sure some angle draw code could be useful for this
>crop = up or down ;this will determine the direction of the crop whether it will crop upwards or downwards assuming the crop has no rotation applied to it meaning the initial position for this is horizontal. I'm sure there is a better more clear terminology for the variables.

>[state 78910, Mask Crop]
>type = maskcrop
>trigger1 = whatever
>value = 4568
>anim = 1234; Not sure if this should use an animation to do the job or just directly access a >sprite to do so

**Comment by LasombraDemon** [comment]
This could also be done via a sprite that would ask a mask for the opponent or for yourself. As an explod.

**Comment by potsmugen** [comment, marked as answer]
Linear cropping at least has been implemented. It existed for explods before and Rakíel added it recently to chars via TransformSprite.
